[MRG] DEP Graph lasso grid scores dep

thomasjpfan opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->
Partly addresses https://github.com/scikit-learn/scikit-learn/issues/11198


#### What does this implement/fix? Explain your changes.
Deprecates `grid_scores_` in `GraphicalLassoCV`

#### Any other comments?
Working toward less issues!

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

thomasjpfan committed:
MNT Deprecates grid_scores_ in favor of cv_results_

thomasjpfan committed:
Merge remote-tracking branch 'upstream/master' into graph_lasso_grid_scores_dep

jnothman commented on sklearn/covariance/_graph_lasso.py:
```diff
@@ -591,6 +592,13 @@ class GraphicalLassoCV(GraphicalLasso):
+            The `grid_scores_` attribute is deprecated in version 0.23 in favor
+            of `cv_results_` and will be removed in version 0.25
+
+    cv_results_ : ndarray of shape (n_alphas, n_folds)
```
Should we return a dict as in parameter search?

adrinjalali replied:
yeah the API closer to what the `gs.cv_results_` looks like would make it easier for people to work with it I guess.

thomasjpfan committed:
CLN Address comments

thomasjpfan committed:
STY Flake8

thomasjpfan committed:
BUG Fix

thomasjpfan committed:
DOC Fixes sphinx warning

adrinjalali reviewed (approved):
otherwise LGTM. Thanks @thomasjpfan

thomasjpfan committed:
Address comments

thomasjpfan committed:
Merge remote-tracking branch 'upstream/master' into graph_lasso_grid_scores_dep

glemaitre commented on sklearn/covariance/_graph_lasso.py:
```diff
@@ -588,9 +589,28 @@ class GraphicalLassoCV(GraphicalLasso):
+            The `grid_scores_` attribute is deprecated in version 0.23 in favor
+            of `cv_results_` and will be removed in version 0.25
+
+    cv_results_ : dict of numpy ndarrays
```
```suggestion
    cv_results_ : dict of ndarrays
```

glemaitre commented on sklearn/covariance/_graph_lasso.py:
```diff
@@ -588,9 +589,28 @@ class GraphicalLassoCV(GraphicalLasso):
+            of `cv_results_` and will be removed in version 0.25
+
+    cv_results_ : dict of numpy ndarrays
+        Dictinoary, with attributes:
```
```suggestion
        Dictionary, with attributes:
```

glemaitre commented on sklearn/covariance/_graph_lasso.py:
```diff
@@ -588,9 +589,28 @@ class GraphicalLassoCV(GraphicalLasso):
+    cv_results_ : dict of numpy ndarrays
+        Dictinoary, with attributes:
+
+        alphas : ndarray of shape (n_alphas,)
```
we have attributes without underscore?

adrinjalali replied:
I think since these are members of the `cv_results_`, they don't need the underscore, do they?

thomasjpfan replied:
I incorrectly wrote "attributes" in the description for `cv_results_`. This has been updated.

glemaitre commented on sklearn/covariance/_graph_lasso.py:
```diff
@@ -767,14 +787,35 @@ def fit(self, X, y=None):
+
+    @deprecated(
+        "The cv_alphas_ attribute is deprecated in version 0.23 in favor "
+        "of cv_results_['alpha'] and will be removed in version 0.25")
```
```suggestion
        "of cv_results_['alpha'] and will be removed in version 0.25"
    )
```

glemaitre commented on sklearn/covariance/_graph_lasso.py:
```diff
@@ -767,14 +787,35 @@ def fit(self, X, y=None):
+
+    @deprecated(
+        "The grid_scores_ attribute is deprecated in version 0.23 in favor "
+        "of cv_results_ and will be removed in version 0.25")
```
```suggestion
        "of cv_results_ and will be removed in version 0.25"
    )
```

glemaitre reviewed (commented):
I will finish my review soonish

thomasjpfan committed:
ENH Adds mean_test_score and std_test_score

thomasjpfan committed:
ENH Adds mean_test_score and std_test_score

thomasjpfan committed:
BUG Fix

thomasjpfan committed:
STY Formating

thomasjpfan commented:
Update PR with:

1. Added `mean_score` and `std_score` keys.
2. Updated `examples/covariance/plot_sparse_cov.py` using the `cv_results_` attribute.

thomasjpfan committed:
STY Flake8

adrinjalali commented on sklearn/covariance/_graph_lasso.py:
```diff
@@ -588,9 +589,34 @@ class GraphicalLassoCV(GraphicalLasso):
+            The `grid_scores_` attribute is deprecated in version 0.23 in favor
+            of `cv_results_` and will be removed in version 0.25
+
+    cv_results_ : dict of ndarrays
```
or a `Bunch`? Either way, I'm happy.

thomasjpfan replied:
I followed how `GridSearchCV` used a dict. We can move them all to `Bunch` in another PR?

adrinjalali reviewed (approved):
Thanks @thomasjpfan

adrinjalali commented:
merge conflict with whats_new.

thomasjpfan committed:
Merge remote-tracking branch 'upstream/master' into graph_lasso_grid_scores_dep

amueller commented:
Is this still waiting for reviews?

thomasjpfan committed:
Merge remote-tracking branch 'upstream/master' into graph_lasso_grid_scores_dep

thomasjpfan committed:
ENH Update to version 0.24

thomasjpfan commented:
Yup waiting for reviews. (Needed update the version to 0.24)

thomasjpfan committed:
ENH Ignore FutureWarning in common test

thomasjpfan committed:
STY Fix

adrinjalali commented:
@glemaitre maybe you'd like to give this another look? ;)

glemaitre committed:
Merge branch 'master' into graph_lasso_grid_scores_dep

glemaitre reviewed (approved):
I resolved the conflicts and I will merge if the CIs are fine.

glemaitre commented:
Thanks @thomasjpfan
